app/src/store/cart.ts

```diff
@@ -95,6 +96,102 @@ export default function useCartStore() {
     }
   }
 
+  // convert the amount to a different token (direction allows for moving to/from the tokenAddress)
+  function getConvertedAmount(amount: number, tokenAddress: string, direction = 1) {
+    const exchangeRate = quotes.value[tokenAddress] ?? 0;
+
+    return amount * (direction == 1 ? exchangeRate : 1 / exchangeRate);
+  }
+
+  // Linear Interpolation function to find the point on the curve that amount corresponds to
+  function lerp(x_lower: number, x_upper: number, y_lower: number, y_upper: number, x: number) {
+    return y_lower + ((y_upper - y_lower) * (x - x_lower)) / (x_upper - x_lower);
+  }
+
+  // returns the prediction curve for this grant in the given round
+  function getPredictionsForGrantInRound(grantId: string, round: GrantRound) {
+    const data = ((grantRoundsCLRData && grantRoundsCLRData.value && grantRoundsCLRData.value) || {}) as {
+      [grantRound: string]: GrantRoundCLR;
+    };
+    const roundData = data[round.address] || {};
+
+    return roundData.predictions && roundData.predictions[Number(grantId)];
+  }
+
+  // given the grantId and the contribution amount find all matching in all rounds
+  function getPredictionForAmount(grantId: string, amount: number, token: TokenInfo) {
+    const rounds = grantRounds.value || [];
+
+    return rounds
+      .map((round) => {
+        const contributions_axis = [0, 1, 10, 100, 1000, 10000];
+        const clr_predictions = getPredictionsForGrantInRound(grantId, round);
+        const clr_prediction_curve = clr_predictions?.predictions?.map((prediction, key) => {
+          return key == 0 ? prediction.predictedGrantMatch : prediction.predictionDiff;
+        });
+
+        let predicted_clr = 0;
+
+        // convert amount to the matchingToken (double hop to get into dai then into the matchingToken)
+        amount =
+          token.symbol == 'dai' || token.address == round.matchingToken.address
```

Comment:
Where are the token symbols populated from? The supported tokens list uses all caps here, and `'dai' == 'DAI'` returns false, so I think we want to change these token equality checks to `token.symbol === 'DAI'` or they'll always return false in that branc.

Alternatively, we really only need the address checks anyway, because the symbol and address checks are equivalent and should either both be true or both be false

Reply:
Ahh, good catch on the casing!! I've refactored and fixed up the naming a little bit to show intent - let me know if theres anything here we need to change